### ISSUE TITLE
could not insert 'btusb': Invalid argument

### ISSUE THREAD

**Issue by ahmedgh970** [tedboudros/tplink-ub500-linux-patch-guide#3]
Hello, 

Thanks a lot for the guide. It's very helpful !

I met an issue with the `sudo modprobe -v btusb` command, though.
```
insmod /lib/modules/5.15.0-52-generic/kernel/drivers/bluetooth/btusb.ko 
modprobe: ERROR: could not insert 'btusb': Invalid argument
```

Any idea ?  Thanks

**Comment by balodi-shashank** [comment]
+1

**Comment by hgrw** [comment]
+1

**Comment by Endstarky** [comment]
Yes that occured to me too but i still continued other steps and rebooted, then after tthat the adapter itself is not recognised by the system now please help!

**Comment by BrelJordan** [comment]
I had the same issue. I had compiled the kernel object for the wrong kernel version a look at dmesg helped me find that out. If you are on ubuntu and "uname -r" gives you the version name :"5.15.0-76-generic" for instance, download the linux kernel "linux-5.15.76.tar.xz" and not "linux-5.15.tar.xz".

After loading the compiled module, Bluetooth tried loading rtl8761bu_fw.bin and rtl8761bu_config.bin instead of rtl8761b_fw.bin and rtl8761b_config.bin. I solved that issue using sim links.
